tool/src/callback_dictionary.ts

```diff
@@ -1,28 +1,130 @@
 import hljs from "highlight.js";
 import mathematica from "highlight.js/lib/languages/mathematica";
+import javascript from "highlight.js/lib/languages/javascript";
+import typescript from "highlight.js/lib/languages/typescript";
+import php from "highlight.js/lib/languages/php";
+import ruby from "highlight.js/lib/languages/ruby";
 
 import { expand, StringObject } from "./util.js"
+import _ from "lodash";
 
-export const callbackDictionary = new Map<string, string>([
-  [
-    hljs.END_SAME_AS_BEGIN({})["on:begin"]!.toString(),
-    "endSameAsBeginOnBegin"
-  ],
-  [
-    hljs.END_SAME_AS_BEGIN({})["on:end"]!.toString(),
-    "endSameAsBeginOnEnd"
-  ],
+const commonCallbacks = new Map<string, string>(
   [
-    hljs.SHEBANG()["on:begin"]!.toString(),
-    "shebangOnBegin",
-  ],
+    [
+      hljs.END_SAME_AS_BEGIN({})["on:begin"]!.toString(),
+      "endSameAsBeginOnBegin"
+    ],
+    [
+      hljs.END_SAME_AS_BEGIN({})["on:end"]!.toString(),
+      "endSameAsBeginOnEnd"
+    ],
+    [
+      hljs.SHEBANG()["on:begin"]!.toString(),
+      "shebangOnBegin",
+    ],
+  ]
+);
+
+export const callbackDictionary = new Map<string, string>([
+  ...commonCallbacks,
 
   ...expand([
-    mathematica,
+    javascript(hljs),
+    mathematica(hljs),
+    php(hljs),
+    ruby(hljs),
+    typescript(hljs),
   ].map(language => getLanguageCallbacks(language))),
 ]);
 
 function getLanguageCallbacks(language: StringObject<any>): [string, string][] {
-  // TODO: Extract all callbacks recursively.
-  return [["a", "b"]];
+  let result: [string, string][] = [];
+
+  const addResult = (k: string, v: string) => {
+    if (!commonCallbacks.has(v.toString())) {
+      result.push([v.toString(), k]);
+    }
+  };
+
+  let objects = <StringObject<any>>[];
+  let languageName: string = language.name;
+  let currentName: string[] = ['k'];
+  if (languageName !== undefined) {
+    currentName.push(languageName.toLowerCase());
+  }
+
+  function iterate(language: StringObject<any>) {
```

Comment:
```suggestion
  function iterate(language: Mode): void {
```

Reply:
If I remove the placeholder `d.ts` file, I get the following message.
Investigating this issue.

Could not find a declaration file for module 'highlight.js'. '/Users/yestay/work/Opensource/dart-highlights/tool/node_modules/highlight.js/es/index.js' implicitly has an 'any' type.
  There are types at '/Users/yestay/work/Opensource/dart-highlights/tool/node_modules/highlight.js/types/index.d.ts', but this result could not be resolved when respecting package.json "exports". The 'highlight.js' library may need to update its package.json or typings.ts(7016)